### PR TITLE
Made SessionData an interface, and introduced DefaultSessionData

### PR DESCRIPTION
**Pull request by Wavesonics**
DefaultSessionData is the implementation previously in SessionData

### CODE REVIEW

File: pippo-session-parent/pippo-session/src/main/java/ro/pippo/session/DefaultSessionData.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
+package ro.pippo.session;
```
Comment (decebals) [inline pippo-session-parent/pippo-session/src/main/java/ro/pippo/session/DefaultSessionData.java]: Please add Copyright header (copy/paste from other file)

File: pippo-session-parent/pippo-session/src/main/java/ro/pippo/session/SessionData.java [hunk 2]
```diff
@@ -28,112 +28,31 @@
  *
  * @author Decebal Suiu
  */
-public class SessionData implements Serializable {
+public interface SessionData extends Serializable {
 
-    public static final int DEFAULT_MAX_INACTIVE_INTERVAL_SECONDS = 30 * 60; // 30 minutes
+    int DEFAULT_MAX_INACTIVE_INTERVAL_SECONDS = 30 * 60; // 30 minutes
```
Comment (decebals) [inline pippo-session-parent/pippo-session/src/main/java/ro/pippo/session/SessionData.java]: Move `DEFAULT_MAX_INACTIVE_INTERVAL_SECOND` in `DefaultSessionData` (I think that is the only place where is used)

File: pippo-session-parent/pippo-session-jdbc/src/main/java/ro/pippo/session/jdbc/JDBCSessionDataStorage.java [hunk 3]
```diff
@@ -24,10 +24,7 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import ro.pippo.core.PippoRuntimeException;
-import ro.pippo.session.SerializationSessionDataTranscoder;
-import ro.pippo.session.SessionData;
-import ro.pippo.session.SessionDataStorage;
-import ro.pippo.session.SessionDataTranscoder;
+import ro.pippo.session.*;
```
Comment (decebals) [inline pippo-session-parent/pippo-session-jdbc/src/main/java/ro/pippo/session/jdbc/JDBCSessionDataStorage.java]: I prefer to not use wildcard import (no `*`)

Comment (decebals) [reply pippo-session-parent/pippo-session-jdbc/src/main/java/ro/pippo/session/jdbc/JDBCSessionDataStorage.java]: They are some files in in this PR that use wildcard import

**Comment by decebals** [comment]
@Wavesonics 
I reviewed the code. I added some comments,

File: pippo-session-parent/pippo-session-jdbc/src/main/java/ro/pippo/session/jdbc/JDBCSessionDataStorage.java (hunk 3 above)
Comment (Wavesonics) [reply pippo-session-parent/pippo-session-jdbc/src/main/java/ro/pippo/session/jdbc/JDBCSessionDataStorage.java]: Oh damn, me too, the IDE must have "optimized" that for me

**Comment by decebals** [comment]
Excellent. Thanks!

**Pull request merged**